org.adempiere.base/src/org/compiere/acct/Doc.java

```diff
@@ -312,6 +313,9 @@ public static String manualPosting (int WindowNo, int AD_Client_ID,
 		catch (Throwable t)
 		{
 			trx.rollback();
+			if(t instanceof SQLException sqlEx) {
+				return Msg.getMsg(Env.getCtx(), DBException.getDefaultDBExceptionMessage(sqlEx));

```

Comment:
this is not ideal as getDefaultDBExceptionMessage can return null.

Reply:
@hengsin
what about something like that?
```java
catch (Throwable t)
		{
			trx.rollback();
                        String error = DBException.getDefaultDBExceptionMessage(sqlEx);
			if(error != null && t instanceof SQLException sqlEx) {
				return Msg.getMsg(Env.getCtx(), error);
			}
			return "@Error@ " + t.getLocalizedMessage();
		}
```

Reply:
'''
if(t instanceof SQLException sqlEx) {
       String error = DBException.getDefaultDBExceptionMessage(sqlEx);
	if(error != null) 
	     return Msg.getMsg(Env.getCtx(), error);
}
'''

Reply:
thanks